conda/plugins/__init__.py

```diff
@@ -5,34 +5,12 @@
 
 import pluggy
 
-from typing import Callable, NamedTuple, Optional
-from collections.abc import Iterable
+from . import specs
+from .specs import CondaSubcommand  # noqa: F401
 
 
-_hookspec = pluggy.HookspecMarker("conda")
 register = pluggy.HookimplMarker("conda")
 
-
-class CondaSubcommand(NamedTuple):
-    """
-    Conda subcommand entry.
-
-    :param name: Subcommand name (e.g., ``conda my-subcommand-name``).
-    :param summary: Subcommand summary, will be shown in ``conda --help``.
-    :param action: Callable that will be run when the subcommand is invoked.
-    """
-    name: str
-    summary: str
-    action: Callable[
-        [list[str]],  # arguments
-        Optional[int],  # return code
-    ]
-
-
-@_hookspec
-def conda_subcommands() -> Iterable[CondaSubcommand]:
-    """
-    Register external subcommands in conda.
-
-    :return: An iterable of subcommand entries.
-    """
+manager = pluggy.PluginManager("conda")
+manager.add_hookspecs(specs)
+manager.load_setuptools_entrypoints("conda")
```

Comment:
FWIW we wanted to make sure that no import-time side effect would trigger the plugin discovery, since it may have unintended downsides to how plugins are loaded. That's why this wasn't in the `conda.plugins` module itself but hidden in a function when the context is loaded during startup. So the way this is written is a risk to add complications when plugins do something weird in their entrypoints or similar.